python/whylogs/core/declarative_schema.py

```diff
@@ -0,0 +1,84 @@
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Mapping, Optional, TypeVar
+
+from whylogs.core.datatypes import DataType, TypeMapper
+from whylogs.core.metrics.metrics import Metric, MetricConfig
+from whylogs.core.resolvers import Resolver
+from whylogs.core.schema import DatasetSchema
+from whylogs.core.segmentation_partition import SegmentationPartition
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class MetricSpec:
```

Comment:
I think MetricSpec and ResolverSpec are general enough to live in another file (maybe resolvers.py?)

Reply:
I'm inclined to either leave things as they are, or move everything but the `DeclarativeSchema` to `resolvers.py`, and `DeclarativeSchema` into `schema.py`. But I'm open to discussion...

Reply:
moving MetricSpec and ResolverSpect to `resolvers.py` and DeclarativeSchema to `schema.py` sounds good to me!
I suggested this because in another PR I'm planning to use MetricSpec and ResolverSpec from `resolvers.py`, and that leads to a circular import.